code/datums/job_configs/job_configs.dm

```diff
@@ -0,0 +1,131 @@
+
+/// This is the proc that we actually invoke to set the config-based values for each job. Is also intended to handle all in-depth logic checks.
+/datum/job_config_type/proc/set_current_value(datum/job/occupation, value)
+	SHOULD_CALL_PARENT(FALSE)
+	CRASH("Attempted to set value for the default job config for [occupation.title] (with config tag [occupation.config_tag])! This is not allowed!")
+
+/// The number of positions a job can have at any given time.
+/datum/job_config_type/default_positions
+	name = JOB_CONFIG_TOTAL_POSITIONS
+
+/datum/job_config_type/default_positions/get_compile_time_value(datum/job/occupation)
+	if(is_assistant_job(occupation)) // yeah i know it's not the "compile time" value but this is just to remain parity with the old system groooan
+		return -1
+	return initial(occupation.total_positions)
+
+/datum/job_config_type/default_positions/validate_value(value)
+	if(isnum(value))
+		return TRUE
+	return FALSE
+
+/datum/job_config_type/default_positions/set_current_value(datum/job/occupation, value)
+	if(!validate_value(value))
+		return FALSE
+	occupation.total_positions = value
+	return TRUE
+
+/// The number of positions a job can have at the start of the round.
+/datum/job_config_type/starting_positions
+	name = JOB_CONFIG_SPAWN_POSITIONS
+
+/datum/job_config_type/starting_positions/get_compile_time_value(datum/job/occupation)
+	if(is_assistant_job(occupation)) // yeah i know it's not the "compile time" value but this is just to remain parity with the old system groooan
+		return -1
+	return initial(occupation.spawn_positions)
+
+/datum/job_config_type/starting_positions/validate_value(value)
+	if(isnum(value))
+		return TRUE
+	return FALSE
+
+/datum/job_config_type/starting_positions/set_current_value(datum/job/occupation, value)
+	if(!validate_value(value))
+		return FALSE
+	occupation.spawn_positions = value
+	return TRUE
+
+/// The amount of playtime required to join a job (minutes).
+/datum/job_config_type/playtime_requirements
+	name = JOB_CONFIG_PLAYTIME_REQUIREMENTS
+
+/datum/job_config_type/playtime_requirements/get_compile_time_value(datum/job/occupation)
+	return initial(occupation.exp_requirements)
+
+/datum/job_config_type/playtime_requirements/validate_value(value)
+	if(isnum(value))
+		return TRUE
+	return FALSE
+
+/datum/job_config_type/playtime_requirements/set_current_value(datum/job/occupation, value)
+	if(!validate_value(value)) // This will mean that the value was commented out, which means that the server operator didn't want to override the codebase default. So, we skip it.
+		return FALSE
+	occupation.exp_requirements = value
+	return TRUE
+
+/// The amount of time required to have an account to join a job (days).
+/datum/job_config_type/required_account_age
+	name = JOB_CONFIG_REQUIRED_ACCOUNT_AGE
+
+/datum/job_config_type/required_account_age/get_compile_time_value(datum/job/occupation)
+	return initial(occupation.minimal_player_age)
```

Comment:
do you think it would be useful to store the linked var name here, and use it automatically (could validate on New())? maybe as a subtype?
A lot of this code looks almost the exact same, which seems inefficient

Reply:
i have no idea what you're talking about here

Reply:
oh i think i know what you mean now, `NAMEOF()` stuff?

Reply:
nope never mind no clue